lib/elixir/lib/stream.ex

```diff
@@ -1404,6 +1401,22 @@ defmodule Stream do
     do: %{lazy | funs: [fun | funs], accs: [acc | accs], done: done}
   defp lazy(enum, acc, fun, done),
     do: %Stream{enum: enum, funs: [fun], accs: [acc], done: done}
+
+  defp do_interval({:first_time, interval}) do
```

Comment:
Though without `do_` this function will clash with the public one.
I think either we should go with anonymous function as @josevalim suggested or perhaps use `interval_iterator` name?